### PR TITLE
Simplify computations of the FrechetMean for the ElasticMetric

### PR DESCRIPTION
**Pull request by ninamiolane**
This PR aimed to simplify the computation of the FrechetMean for the ElasticMetric, and ended up solving a series of bugs on elastic curves.

The method for the FrechetMean on elastic curves works as follows:
- transforming the data point through the f transform
- computing their linear mean
- transforming the mean back into the curve space with the f transform inverse.

This PR also:
- adds better docstrings for the SRVMetric and the ElasticMetric
- brings back unit-tests about the f_transform and polar coordinates that were deleted
- vectorize the computations of the functions cartesian <> polar coordinates , and f_transform with its inverse
- adds corresponding unit-tests

## Checklist

- [x] My pull request has a clear and explanatory title.
- [x] If neccessary, my code is [vectorized](https://www.geeksforgeeks.org/vectorization-in-python/).
- [x] I have added apropriate unit tests.
- [ ] I have made sure the code passes all unit tests. (refer to comment below)
- [x] My PR follows [PEP8](https://peps.python.org/pep-0008/) guidelines. (refer to comment below)
- [x] My PR follows [geomstats coding style](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#coding-style-guidelines) and API.
- [x] My code is properly documented and I made sure the documentation renders properly. ([Link](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#documentation))
- [x] I have linked to issues and PRs that are relevant to this PR.

### CODE REVIEW

**Comment by ninamiolane** [comment]
The failing tests are not related to this PR.

File: geomstats/geometry/discrete_curves.py [hunk 1]
```diff
@@ -464,7 +465,7 @@ def __init__(
                     " equipped with a metric."
                 )
         self.ambient_manifold = ambient_manifold
-        self.l2_metric = L2CurvesMetric(ambient_manifold=ambient_manifold)
+        self.l2_curve_metric = L2CurvesMetric(ambient_manifold=ambient_manifold)
```
Comment (alebrigant) [inline geomstats/geometry/discrete_curves.py]: Maybe rename to `l2_curves_metric` for coherence, or `L2CurvesMetric` to `L2CurveMetric`?

File: geomstats/geometry/discrete_curves.py [hunk 2]
```diff
@@ -670,7 +671,7 @@ def f_transform_inverse(self, f, starting_point):
             )
         if gs.ndim(f) != gs.ndim(starting_point):
             starting_point = gs.to_ndarray(starting_point, to_ndim=f.ndim, axis=-2)
-        self.l2_metric
+        self.l2_curve_metric
```
Comment (alebrigant) [inline geomstats/geometry/discrete_curves.py]: What does this line do ? I don't see it in the current version of the master branch. For some reason, the current version used to compare changes in this PR does not seem to correspond to the current version of the master branch.

File: geomstats/geometry/discrete_curves.py [hunk 3]
```diff
@@ -724,39 +725,55 @@ def dist(self, curve_1, curve_2, rescaled=False):
...
+        f_1 = self.polar_to_cartesian(f_1)
+        f_2 = self.polar_to_cartesian(f_2)
 
         if rescaled:
-            cosine = l2_prod(f_1, f_2) / (4 * self.b**2)
+            cosine = self.l2_curve_metric.inner_product(f_1, f_2) / (4 * self.b**2)
             distance = 2 * self.b * gs.arccos(gs.clip(cosine, -1, 1))
         else:
-            distance = l2_dist(f_1, f_2)
+            # elastic_metric.dist
```
Comment (alebrigant) [inline geomstats/geometry/discrete_curves.py:768]: Leftover comments ?

**Review by alebrigant — COMMENTED** [review]
Looks good to me ! I may be missing something, but the version to which the changes are compared does not seem to be the current version of the master branch.

File: geomstats/geometry/discrete_curves.py (hunk 1 above)
Comment (ninamiolane) [reply geomstats/geometry/discrete_curves.py]: Yes, done!

File: geomstats/geometry/discrete_curves.py (hunk 2 above)
Comment (ninamiolane) [reply geomstats/geometry/discrete_curves.py]: This was a left over,thanks for catching it

File: geomstats/geometry/discrete_curves.py (hunk 3 above)
Comment (ninamiolane) [reply geomstats/geometry/discrete_curves.py:768]: yes, thanks

**Pull request merged**